flutter/news/lib/features/frontpage/data/repositories/articles_repository_impl.dart

```diff
@@ -1,19 +1,46 @@
 import 'package:dartz/dartz.dart';
-import 'package:news/core/error/failures.dart';
-import 'package:news/features/frontpage/domain/entities/article.dart';
-import 'package:news/features/frontpage/domain/repositories/articles_repository.dart';
+import 'package:news/core/error/exceptions.dart';
+
+import '../../../../core/error/failures.dart';
+import '../../../../core/network/network_info.dart';
+import '../../domain/entities/article.dart';
+import '../../domain/repositories/articles_repository.dart';
+import '../datasource/articles_data_source.dart';
+import '../datasource/articles_local_data_source.dart';
 
 class ArticlesRepositoryImpl implements ArticlesRepository {
+  final ArticlesLocalDataSource localDataSource;
+  final ArticlesRemoteDataSource remoteDataSource;
+  final NetworkInfo networkInfo;
+
+  ArticlesRepositoryImpl(
+      {required this.localDataSource,
+      required this.remoteDataSource,
+      required this.networkInfo});
+
   @override
   Future<Either<Failure, List<Article>>> getEverythingAbout(String query) {
     // TODO: implement getEverythingAbout
     throw UnimplementedError();
   }
 
   @override
-  Future<Either<Failure, List<Article>>> getTopHeadlines() {
-    // TODO: implement getTopHeadlines
-    throw UnimplementedError();
+  Future<Either<Failure, List<Article>>> getTopHeadlines() async {
```

Comment:
Does something like `Result` exists in `dart` that we can use instead of `either`?

Reply:
A `Result` looks like an `either` but it doesn't have `left` or `right`

A `Result` has` Success(data)` and `Failure` (which you have already implemented as an abstract class)